src/std/os/signal-handler.ss

```diff
@@ -98,7 +93,93 @@ package: std/os
        (sigaddset sigset signo)
        (sigprocmask SIG_UNBLOCK sigset #f))))
 
-  ;; TODO bsd -- kqueue implementation
+  (bsd
+
+   (defstruct signal-handler (handlers default-handlers kq mx)
+     final: #t constructor: :init!)
+
+   (defmethod {:init! signal-handler}
+     (lambda (self)
+       (let* ((handlers (make-vector SIGMAX #f))
+              (default-handlers (make-vector SIGMAX #f))
+              (kq (kqueue))
+              (mx (make-mutex 'signal-handler)))
+         (struct-instance-init! self handlers default-handlers kq mx)
+         (start-logger!)
+         (spawn/name 'signal-handler signal-handler-wait self))))
+
+   (def (signal-handler-wait sh)
+     (try
+      (defconst nevents 4)
+      (def events (make-kevents nevents))
+      (with ((signal-handler handlers default-handlers kq mx) sh)
+        (let wait-loop ()
+          (##wait-for-io! (fd-io-in kq) #t)
+          (let (n (kqueue-poll kq events nevents))
+            (when (##fxpositive? n)
+              (let event-loop ((i 0))
+                (when (fx< i n)
+                  (if (##fxzero? (##fxand (kevent-flags events i)
+                                          EV_ERROR))
+                    (when (fx= (kevent-filter events i) EVFILT_SIGNAL)
+                      (let (event-handler
+                            (with-lock mx
+			      (cut vector-ref handlers
```

Comment:
this is still not correctly indented -- it should look like this:
```
(with-lock mx
  (cut vector-ref ...))
```

Add it to the indentation table with `M-x eval (gerbil-put-indent '(with-lock) 1)` and let emacs indent the file!

Reply:
I don't know why this happens. I did the indentation with emacs. I try a second time.

Reply:
Now I know the reason: there are tabs at the beginning of the line. I will correct it.